src/supplemental/http/http_server.c

```diff
@@ -560,17 +560,21 @@ http_sconn_rxdone(void *arg)
 		if (strncmp(path, h->uri, len) != 0) {
 			continue;
 		}
-		switch (path[len]) {
-		case '\0':
-			break;
-		case '/':
-			if ((path[len + 1] != '\0') && (!h->tree)) {
-				// Trailing component and not a directory.
-				continue;
+		if (len == 1 && h->uri[0] == '/') {

```

Comment:
This code is logically not right.

You're bypassing the entire tree check here for one.

What is the value of h->uri here and path here?